### PR TITLE
fix: reduce activation time

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: packages/backend/src/panels/AbstractWebviewPanel.ts [hunk 1]
```diff
@@ -99,7 +99,7 @@ export abstract class AbstractWebviewPanel {
     if (this.flowPromise) {
       // resolves promise in case panel is closed manually by an user
       // it is safe to call resolve several times on same promise
```
Comment (alex-gilin) [inline packages/backend/src/panels/AbstractWebviewPanel.ts:101]: this.flowPromise?.state?.resolve()

File: packages/backend/src/panels/ExploreGensPanel.ts [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
 import { getWebviewRpcLibraryLogger } from "../logger/logger-wrapper";
-import _ = require("lodash");
+import { get, isNil } from "lodash";
 
 export class ExploreGensPanel extends AbstractWebviewPanel {
-  public setWebviewPanel(webviewPanel: vscode.WebviewPanel, uiOptions?: unknown) {
+  public constructor(context: Partial<ExtensionContext>) {
+    super(context);
+    this.viewType = "exploreGens";
+    this.viewTitle = "Explore and Install Generators";
```
Comment (alex-gilin) [inline packages/backend/src/panels/ExploreGensPanel.ts:13]: exclude to messages ?

File: packages/backend/src/panels/ExploreGensPanel.ts [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+    this.focusedKey = "exploreGens.Focused";
+    this.htmlFileName = join("exploregens", "index.html");
+    this.exploreGens = new ExploreGens(this.logger, this.isInBAS, this.context);
+  }
+
+  public setWebviewPanel(webviewPanel: WebviewPanel, uiOptions?: unknown) {
```
Comment (alex-gilin) [inline packages/backend/src/panels/ExploreGensPanel.ts:19]: uiOptions - unused, isn't ?

File: packages/backend/src/panels/ExploreGensPanel.ts [hunk 4]
```diff
@@ -16,7 +25,7 @@ export class ExploreGensPanel extends AbstractWebviewPanel {
   }
 
   private async installGenOnPanelOpenIfNeeded(uiOptions?: unknown) {
-    const genFullName = _.get(uiOptions, "package.name");
+    const genFullName = get(uiOptions, "package.name");
     if (genFullName) {
       try {
         await this.exploreGens.setGenFilter(genFullName);
```
Comment (alex-gilin) [inline packages/backend/src/panels/ExploreGensPanel.ts:31]: await at next line is redundant

File: packages/backend/src/panels/ExploreGensPanel.ts [hunk 5]
```diff
@@ -30,17 +39,8 @@ export class ExploreGensPanel extends AbstractWebviewPanel {
...
-    this.htmlFileName = path.join("exploregens", "index.html");
-    this.exploreGens = new ExploreGens(this.logger, this.isInBAS, this.context);
-  }
-
   public async loadWebviewPanel(uiOptions?: unknown) {
-    if (this.webViewPanel && _.isNil(uiOptions)) {
+    if (this.webViewPanel && isNil(uiOptions)) {
       this.webViewPanel.reveal();
     } else {
       await super.loadWebviewPanel(uiOptions);
```
Comment (alex-gilin) [inline packages/backend/src/panels/ExploreGensPanel.ts:46]:
```suggestion
      super.loadWebviewPanel(uiOptions);
```

File: packages/backend/src/panels/YeomanUIPanel.ts [hunk 6]
```diff
@@ -133,14 +133,14 @@ export class YeomanUIPanel extends AbstractWebviewPanel {
 
     let uri;
     try {
-      if (_.isEmpty(currentPath)) {
+      if (isEmpty(currentPath)) {
         throw new Error("Empty path");
       }
       uri = vscode.Uri.file(currentPath);
     } catch (e) {
```
Comment (alex-gilin) [inline packages/backend/src/panels/YeomanUIPanel.ts:140]: uri = vscode.Uri.file(currentPath);
if (!uri?.fsPath) {
    throw new Error("Empty path");
}

File: packages/backend/src/utils/vscodeProxy.ts [hunk 7]
```diff
@@ -13,17 +13,17 @@ export const getVscode = () => {
 const filename: string = require.main.filename;
 const _isInTest = filename.includes(join("node_modules", "mocha"));
 
-const notImplementedError = new Error("tested method is not implemented");
-const returnValue = () => {
+//const notImplementedError = new Error("tested method is not implemented");
```
Comment (alex-gilin) [inline packages/backend/src/utils/vscodeProxy.ts:16]: remove ?

**Review by alex-gilin — CHANGES_REQUESTED** [review]

File: packages/backend/src/utils/vscodeProxy.ts (hunk 7 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/vscodeProxy.ts]: Done

File: packages/backend/src/panels/ExploreGensPanel.ts (hunk 3 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/ExploreGensPanel.ts:19]: they are used in this method

File: packages/backend/src/panels/ExploreGensPanel.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/ExploreGensPanel.ts:13]: what messages ?

File: packages/backend/src/panels/ExploreGensPanel.ts (hunk 5 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/ExploreGensPanel.ts:46]: no, this method is async and we need to wait till it finishes

File: packages/backend/src/panels/YeomanUIPanel.ts (hunk 6 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/YeomanUIPanel.ts:140]: I do not understand what how it helps ?
Please explain

File: packages/backend/src/panels/ExploreGensPanel.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/ExploreGensPanel.ts]: Done

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**